### ISSUE TITLE
[BEHAVIOR] Improvements in fitting strategy

### ISSUE THREAD

**Issue by acampove** [zfit/zfit#588]
## Describe what happens

Fits give nonsensical answers when the datasets get too large. There are two scenarios:

### Large datasets
When we fit large datasets, the fitter seems to get confused and pick minima that do not make sense.

#### Solution

Do what RooFit does and fit a small subset of the data to find a set of starting values for the parameters

### Complex models

The same as before. When the model has a lot of parameters, the fit just fails. This is mostly a problem with MC fits. The usecase is to get the tails.

#### Solution

Implement a fit manager class that will retry the fit multiple times with randomized initial values and pick the fit with the best GOF.

**Comment by acampove** [comment]
In any case, I have private code to do this and I am gradually moving it to a public repository. You can find what I am doing here:

https://github.com/acampove/dmu/tree/main?tab=readme-ov-file#fits

again, you can pick whatever you need from there.

**Comment by ikrommyd** [comment]
@jonas-eschle This sounds like a `CascadeMinimizer` kind of request which combine implements. I also recently requested for a more robust under the hood fit in iminuit which was implemented here: https://github.com/scikit-hep/iminuit/pull/1009

**Comment by jonas-eschle** [comment]
Yes, and I think it's something that is easily achievable with the BaseMinimizer that we have in the tutorial, i.e. running multiple minimizers and seeing which one returns the lowest loss, basically
